### DISCUSSION TITLE
13 May 2024 1145 US-Eastern

### DISCUSSION THREAD

**Discussion by ktoddbrown** [ktoddbrown/SoilDRaH#10] [RegularCalls]
**Recap**
In this first meeting of the 2024 semester, we introduced our new github-based notes method, looked back at todo items from previous semesters, and planned to do some planning/design consolidation using github project management features. Maggie and Gabriel will work on a workflow diagram, Graham will write some unit tests, and Dr. Todd-Brown and Vaasuki will work with the FIA and CPEAT branches respectively.

**Agenda**:
- Check ins: Who is here? Note taker? How are you showing up?
- Orientation to GitHub Discussions and new notes
- Planning summer work
- Check out: What are you leaving with [appreciation, ToDo, curiosity]

**Comment by graham-louis** [comment]
### Meeting Notes
- **Check ins**: 
- Who is here? Maggie J, Graham L, Gabriel T, Vaasuki M. 
- Note taker? Graham L.
-  How are you showing up? -no notes-
- **Orientation to GitHub Discussions and new notes**
  - Instead of google doc we will use Discussions feature in GitHub
  - Try to keep organization more publicly oriented
  - **Reactions**
    - Maggie: Likes that things are in one spot
    - Gabriel: More streamlined
    - Vaasuki: Still concerned about missing org document (One of Dr. TB’s motivations for switching). Will send slack to Ursa to check on status.
  - **Workflow**
    - Notetaker should download notes document from zoom as backup.
    - Copy and paste notes into discussion item within GitHub under a notes header, add recap section also
- **Planning summer work**
  - Outstanding To-do items from repo
    - Deliverables timeline?
      - Under Projects/ISCN4 in repo
      - Doesn’t have dates
      - Open items under Issues tab
    - Some unit tests that need to be created
    - Data ingest things
      - 4 different sources: NRCS, ISCN3, FS-FIA, and CPEAT
        - CPEAT and FS-FIA are under work
        - ISCN3 is a large task
        - NRCS-NCSS: Annotation need to be finalized
    - Get all 4 sources finished by end of summer
      - Annotation checks: Verify that they’re correct
      - Transparently document merges
      - Get workflow well documented so that a user has a clear guide for how to merge a new data set
    - Vaasuki: Design elements should be finalized before annotation workflow is continued
      - Issue with zip file for CPEAT
      - Explicitly map out workflow
      - Finalize annotations (QA/QC)
        - Call primary data contributors and...
        - First step should be check annotation functions then...
        - Send a report to contributors
      - Formalize and document design decisions within repo
      - Build out time line from core goals
  - **Curiosities?**
    - Workshop was cancelled => Looser timeline
      - Still want to finish up ISCN4 by end of Summer
    - Todo item structuring
      - Task assignments? Deadlines?
        - This is a good thing to figure out immediately
        - Fork repo and clone to local machine
        - On feature branch structure...
          - One branch for each source + one for check-annotation function
          - Using testthat suite for unit testing of check-annotation
  - **First steps**
    - Commit data annotations for CPEAT
    - Data annotations are included in readCPEAT.R, however...
      - annotations should be moved over to data folder once pipeline is debugged
  - Gabriel and Maggie => Create a workflow diagram for downloading, reading, and annotating.
    - Use power point and focus on content
    - Starting with initial host, what his happening to bring the data product into ISCN4
    - Focus on big picture; how does it all fit together
    - start with vignettes/readTemplate.Rmd
  - Graham => Unit tests for checkAnnotations, add min/max from an annotations table
    - Work on 8-check-annotations branch???
  - Vaasuki => Will focus on CPEAT first
  - Dr. TB => do a check in with FIA
  - Get annotations for both of these^ ported to their respective branches
    - Maggie and Gabriel will use this for their workflow diagram
    - Vaasuki volunteers her help with git things
- **Talk about next time**
  - Check in todo items with stuck to unstuck
  - Milestones and timelines for summer goals!
  - Statement of purpose for mine ReadMe.md
  - Design document - via wiki!
- **Check out**: What are you leaving with [appreciation, ToDo, curiosity]

### Recap
In this first meeting of the 2024 semester, we introduced our new github-based notes method, looked back at todo items from previous semesters, and planned to do some planning/design consolidation using github project management features. Maggie and Gabriel will work on a workflow diagram, Graham will write some unit tests, and Dr. Todd-Brown and Vaasuki will work with the FIA and CPEAT branches respectively.
